core/client/fs/src/test/java/alluxio/client/file/cache/LocalCacheManagerTest.java

```diff
@@ -819,14 +822,27 @@ public void asyncCache() throws Exception {
     pageStore.setPutHanging(true);
     mPageMetaStore = new DefaultPageMetaStore(ImmutableList.of(dir));
     mCacheManager = createLocalCacheManager(mConf, mPageMetaStore);
+    Set<Integer> lockedPages = new HashSet<>();
     for (int i = 0; i < threads; i++) {
       PageId pageId = new PageId("5", i);
       assertTrue(mCacheManager.put(pageId, page(i, PAGE_SIZE_BYTES)));
+      lockedPages.add(mCacheManager.getPageLockId(pageId));
```

Comment:
Does it need to assertTrue so that those page ids will not be conflicted

Reply:
Here it is fine if the worker threads conflict on the same lock, since they are async it is fine if the block each other. We just don't want the main thread to be waiting on a lock owned by a worker thread, since then the main thread will reach the point where it can disable hanging.